### PR TITLE
[KYUUBI #939][FOLLOWUP] Refine pom.xml

### PR DESCRIPTION
**Pull request by pan3793**
### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.readthedocs.io/en/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: dev/kyuubi-extension-spark-3-1/pom.xml [hunk 1]
```diff
@@ -104,33 +104,21 @@
         <dependency>
             <groupId>javax.xml.bind</groupId>
             <artifactId>jaxb-api</artifactId>
-        </dependency>
-
-        <dependency>
-            <groupId>org.antlr</groupId>
-            <artifactId>antlr4-runtime</artifactId>
+            <scope>test</scope>
```
Comment (pan3793) [inline dev/kyuubi-extension-spark-3-1/pom.xml:107]: it's like an accident change in #990, reverted.

File: dev/kyuubi-extension-spark-3-1/pom.xml [hunk 2]
```diff
@@ -104,33 +104,21 @@
         <dependency>
             <groupId>javax.xml.bind</groupId>
             <artifactId>jaxb-api</artifactId>
-        </dependency>
-
-        <dependency>
-            <groupId>org.antlr</groupId>
-            <artifactId>antlr4-runtime</artifactId>
```
[the remark this answers is not in the document]
Comment (ulysses-you) [reply dev/kyuubi-extension-spark-3-1/pom.xml:111]: we can also remove the `antlr4-runtime` and parent pom.

[the remark this answers is not in the document]
Comment (cfmcgrady) [reply dev/kyuubi-extension-spark-3-1/pom.xml:111]: Just make sure we don't miss anything here,  Spark-3.1.2 depends `antlr4-runtime:4.8-1`, but kyuubi generates source code with `antlr-maven-plugin:4.7`. Is there any compatibility issue?

https://github.com/apache/spark/blob/de351e30a90dd988b133b3d00fa6218bfcaba8b8/pom.xml#L192

```xml
    <antlr4.version>4.8-1</antlr4.version>
```

https://github.com/apache/incubator-kyuubi/blob/4cd83790ae651ee0da92365458b08224d68a6419/pom.xml#L90

[the remark this answers is not in the document]
Comment (pan3793) [reply dev/kyuubi-extension-spark-3-1/pom.xml:111]: Yeah, we should bump `4.8-1` to align with spark, `4.7` should work but will print warning messages if the antlr version is not exactly matched.

[the remark this answers is not in the document]
Comment (pan3793) [reply dev/kyuubi-extension-spark-3-1/pom.xml:111]: Actually, we should bump antlr 4.8, see https://github.com/apache/spark/pull/32603

[the remark this answers is not in the document]
Comment (cfmcgrady) [reply dev/kyuubi-extension-spark-3-1/pom.xml:111]: make sense

**Review by ulysses-you — APPROVED** [review]

**Comment by ulysses-you** [comment]
thanks, merging to master

**Pull request closed without merging**